packages/core/src/connectors/walletConnect.ts

```diff
@@ -56,6 +73,58 @@ export class WalletConnectConnector extends Connector<
       provider.on('chainChanged', this.onChainChanged)
       provider.on('disconnect', this.onDisconnect)
 
+      if (this.version === '2') {
+        provider.on('session_delete', this.onDisconnect)
+        provider.on('display_uri', this.onDisplayUri)
+
+        await Promise.race([
+          provider.connect({
+            namespaces: {
+              [sharedConfig.namespace]: {
+                methods: [
+                  'eth_sendTransaction',
+                  'eth_sign',
+                  'eth_signTransaction',
+                  'eth_signTypedData',
+                  'personal_sign',
+                ],
+                events: ['accountsChanged', 'chainChanged'],
+                chains: this.chains.map(
+                  (chain) => `${sharedConfig.namespace}:${chain.id}`,
+                ),
+                rpcMap: this.chains.reduce(
+                  (rpc, chain) => ({
+                    ...rpc,
+                    [chain.id]: chain.rpcUrls.default,
+                  }),
+                  {},
+                ),
+              },
+            },
+          }),
+          ...(this.options.qrcode
+            ? // When using WalletConnect QR Code Modal, open modal and listen for close callback.
+              // If modal is closed, reject promise so `catch` block for `connect` is called.
+              [
+                new Promise((_res, reject) =>
+                  provider.on('display_uri', async (uri: string) => {
+                    const { default: Modal } = await import(
+                      '@walletconnect/qrcode-modal'
+                    )
+                    Modal.open(uri, () => reject(new Error('user rejected')))
+                  }),
+                ),
+              ]
+            : []),
```

Comment:
Need this block so when user closes out of QR Code Modal, "rejected request" error bubbles up to user.